docs/articles/templates/selectors/selector-options.md

```diff
@@ -0,0 +1,105 @@
+## Options
+
+### options.boundTestRun
+
+**Type**: Object
+
+If you need to call a selector from a Node.js callback, assign the current [test controller](../../test-code-structure.md#test-controller) to the `boundTestRun` option.
+
+For details, see [Calling Selectors from Node.js Callbacks](edge-cases-and-limitations.md#calling-selectors-from-nodejs-callbacks).
+
+### options.dependencies
+
+**Type**: Object
+
+Use this option to pass functions, variables or objects to selectors [initialized with a function](creating-selectors.md#initialize-selectors). The `dependencies` object's properties are added to the function's scope as variables.
+
+Use `dependencies` instead of the function's arguments if you do not need to pass new values every time you call the selector.
+
+The following sample demonstrates a selector (`element`) that uses a server-side object passed as a dependency (`customId`) to obtain a page element.
+
+```js
+import { Selector } from 'testcafe';
+
+const persistentId = { key: 'value' };
+
+const element = Selector(() => {
+    return getElementByCustomId(persistentId);
+}, {
+    dependencies: { persistentId }
+});
+
+fixture `My fixture`
+    .page `http://www.example.com/`;
+
+test('My Test', async t => {
+    await t.click(element);
+});
+```
+
+### options.timeout
+
+**Type**: Number
+
+The amount of time, in milliseconds, allowed for an element returned by the selector to appear in the DOM before the test fails.
+
+If the [visibilityCheck](#optionsvisibilitycheck) option is enabled, the element then must become visible within the `timeout`.
+
+**Default value**: the timeout specified in the [runner.run](../../../using-testcafe/programming-interface/runner.md#run) API method or the [--selector-timeout](../../../using-testcafe/command-line-interface.md#--selector-timeout-ms) command line option.
+
+### options.visibilityCheck
+
+**Type**: Boolean
+
+`true` to additionally require the returned element to become visible within [options.timeout](#optionstimeout).
+
+This option is in effect when TestCafe waits for the selector to return a page element. This includes situations when
+
+* a property is obtained from the selector;
+
+    ```js
+    const width = await Selector('#element', { visibilityCheck: true }).clientWidth;
+    ```
+
+* a selector property is passed to an [assertion](../../assertions/README.md) as its actual value;
+
+    ```js
+    await t.expect(Selector('#element', { visibilityCheck: true }).clientWidth).eql(400);
+    ```
+
+* a selector is evaluated with the `await` keyword;
+
+    ```js
+    const snapshot = await Selector('#element', { visibilityCheck: true })();
+    ```
+
+If the target element is not visible, the selector throws an exception in all these cases.
```

Comment:
Is `in all these cases.` needed here?  If we need to highlight that it related to only the situations listed above, maybe it's possible to rephrase this to make it a bit clearer?